archetypes/archetype-lambda/src/test/resources/projects/nettyclient/reference/src/main/java/software/amazonaws/test/DependencyFactory.java

```diff
@@ -0,0 +1,26 @@
+
+package software.amazonaws.test;
+
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient;
+
+/**
+ * The module containing all dependencies required by the {@link MyNettyFunction}.
+ */
+public class DependencyFactory {
+
+    private DependencyFactory() {}
+
+    /**
+     * @return an instance of KinesisAsyncClient
+     */
+    public static KinesisAsyncClient kinesisClient() {
+        return KinesisAsyncClient.builder()
+                       .credentialsProvider(EnvironmentVariableCredentialsProvider.create())
+                       .region(Region.US_EAST_1)
+                       .httpClientBuilder(NettyNioAsyncHttpClient.builder())
```

Comment:
Do we include open SSL or other tweaks for netty?

Reply:
That would require another openssl dependency `netty-tcnative` though, and not all platforms support it. We can probably add a comment here to recommend using openssl if possible.

Reply:
But since it's on Lambda, we know the platform, right?

Reply:
lol, good point. I'll give it a try

Reply:
Wow, the `netty-tcnative-boringssl-static` binding jar itself is 3MB...Certainly wouldn't help with cold start issue, but I guess it would improve overall latency a bit.

Reply:
Ended up just adding `netty-tcnative-boringssl-static` dependency so that it'd use openssl if supported and fallback to jdk ssl provider otherwise.